src/libraries/System.Net.Quic/tests/FunctionalTests/MsQuicTests.cs

```diff
@@ -30,10 +30,10 @@ public async Task UnidirectionalAndBidirectionalStreamCountsWork()
         {
             using QuicListener listener = CreateQuicListener();
             using QuicConnection clientConnection = CreateQuicConnection(listener.ListenEndPoint);
+            Task<QuicConnection> serverTask = listener.AcceptConnectionAsync().AsTask();
```

Comment:
SInce this is a common, repeated pattern, I wonder if we should have a helper method for it, like CreateClientAndServer or something like that. Or EstablishConnection or something. 

We have RunClientServer, but that's not really the same.

Reply:
yah. I wanted to like SslStream. But if I return tuple we loose the `using var`.
We would need to than use good old
```c#
using (clientConn)
using (serverConn)
{
  ....
}
```

Reply:
BTW the helper would be also nice _if_ we want to bet in any retry logs for the flaky listener if we don't find way how to fix it.

Reply:
Yeah, it's a little verbose because of the using stuff.

One idea would be to define something like

```csharp
struct QuicConnectionPair : IDisposable
{
  public QuicConnection Client { get; }
  public QuicConnection Server { get; }
  public void Dispose() { Client.Dispose(); Server.Dispose(); }
}

public static ValueTask<ConnectionPair> EstablishConnectionAsync(...) { ... }
```

Then you can just write code like this:

```csharp
  using (var connectionPair = await EstablishConnectionAsync(...);
  // use connectionPair.Client and connectionPair.Server here
```

If we feel really ambitious we could generalize `ConnectionPair` to something like `DisposablePair<T1, T2>`. And maybe add some implicit conversion ops from `ValueTuple<T1, T2>`...

@stephentoub already has something vaguely like this for the Stream Conformance Tests.